16/umbraco-cms/customizing/foundation/context-api/context-api-fundamentals.md

```diff
@@ -0,0 +1,74 @@
+A unique identifier used to request specific contexts. Context tokens serve as contracts between providers and consumers. They define exactly which context is being requested and ensure that the right provider responds. Using a context token prevents conflicts when multiple contexts might have similar names and makes clear what functionality is being shared.
+
+## Context consume flow
+Each DOM element can be a context provider. Each descendant in the DOM hierarchy can consume that context if desired. When an element wants to consume a context, the following happens:
+
+1. An element requests a context by a given Context Token.
+2. The Context API dispatches an event that starts at the element that requested the context. The event bubbles up the DOM tree to each parent element until an element is found that responds to the event.
+3. An instance of the context is provided back to the element that requested the context.
+
+![Context API Flow](images/umbraco_context_api_flow.png)
+
+If no context could be found and the event reaches the top-level element (the document), no context is consumed.
+
+## Common contexts
+Although every element can be a context provider, the most important contexts are registered at specific hierarchy levels. These levels are also explicit extension points in the Umbraco manifest. 
+
+The hierarchy levels to which contexts can be registered are:
+
+* Global
+* Section
+* Workspace
+* Property
+
+**Global contexts** are registered at the highest level and are always available anywhere in the backoffice. Examples of global contexts:
+* `Notification context`: used for displaying notifications in the backoffice. This context is consumable in elements anywhere in the DOM tree.
+* `Current user context`: has information about the currently logged in user. This context is consumable anywhere in the DOM tree.
+
+**Section contexts** are available in the context of a section. That's everything in the backoffice except the menubar. Examples of section contexts:
+* `Section context`: provides information about the section, like path, alias, and label.
+* `Sidebar menu section context`: holds information about the sidebar menu, like which menu is currently selected.
+
+**Workspace contexts** work on a workspace; the part of Umbraco that's next to the tree. Example of this level:
+* `Workspace context`: holds information about the current entity being edited in the workspace. This holds minimal information about an entity and the entity type. There are specific workspace contexts per entity type. For instance, the `Document workspace context` for documents and `Media workspace context` for media.
+
+**Property contexts** are contexts that work at the property level. They can work on one or more property editors. An example is the clipboard functionality where blocks can be copied and pasted between block grids and block lists. Because these contexts are scoped at the property level, they are typically not consumed directly.
```

Comment:
```suggestion
**Property contexts** are contexts that work at the property level. They can work on one or more property editors. An example is the clipboard functionality, where blocks can be copied and pasted between block grids and block lists.   Because these contexts are scoped at the property level, they are typically not consumed directly.
```